File: Todo.UI/src/types/comment.ts
import User from "@/types/user";

export type Comment = {
  id: string;
  author: User;
  text: string;
  creationTimeStamp: string;
  updateTimeStamp: string;
};

export type CreateComment = {
  itemId: string;
  text: string;
};

export type EditComment = {
  id: string;
  itemId: string;
  text: string;
};

export type DeleteComment = {
  id: string;
  itemId: string;
};

File: Todo.Domain/Entities/Comment.cs
namespace Todo.Domain.Entities;

public class Comment : Entity
{
    public Guid AuthorId { get; set; }
    public virtual User Author { get; set; }
    public string Text { get; set; }
    public DateTime CreationTimeStamp { get; set; }
    public DateTime UpdateTimeStamp { get; set; }
    public Guid ItemId { get; set; }
    public virtual TodoItem Item { get; set; }
    
    public Comment(Guid authorId, Guid itemId, string text)
    {
        AuthorId = authorId;
        ItemId = itemId;
        Text = text;
        CreationTimeStamp = DateTime.UtcNow;
        UpdateTimeStamp = DateTime.UtcNow;
    }
}